consensus/src/consensus/storage.rs

```diff
@@ -74,91 +74,121 @@ impl ConsensusStorage {
         )));
+
+        let max_ghostdag_data_size = size_of::<GhostdagData>()
+            + params.mergeset_size_limit as usize * size_of::<Hash>()
+            + params.ghostdag_k as usize * size_of::<(Hash, KType)>();
+
         let ghostdag_stores = Arc::new(
             (0..=params.max_block_level)
                 .map(|level| {
-                    let cache_size = max(ghostdag_cache_size.checked_shr(level as u32).unwrap_or(0), 2 * params.pruning_proof_m);
-                    Arc::new(DbGhostdagStore::new(db.clone(), level, noise(cache_size)))
+                    // = size / 2^level
+                    let level_normalized_cache_bytes = ghostdag_cache_bytes.checked_shr(level as u32).unwrap_or(0);
+                    let cache_policy = if level_normalized_cache_bytes > 2 * params.pruning_proof_m as usize * max_ghostdag_data_size {
+                        CachePolicy::Tracked(noise(level_normalized_cache_bytes))
+                    } else {
+                        CachePolicy::Unit(noise(2 * params.pruning_proof_m as usize))
+                    };
+                    let compact_cache_size =
+                        max(ghostdag_compact_cache_size.checked_shr(level as u32).unwrap_or(0), 2 * params.pruning_proof_m as usize);
+                    Arc::new(DbGhostdagStore::new(db.clone(), level, cache_policy, CachePolicy::Unit(noise(compact_cache_size))))
                 })
                 .collect_vec(),
         );
         let ghostdag_primary_store = ghostdag_stores[0].clone();
-        let daa_excluded_store = Arc::new(DbDaaStore::new(db.clone(), noise(daa_excluded_cache_size)));
-        let headers_store = Arc::new(DbHeadersStore::new(db.clone(), noise(perf_params.headers_cache_size)));
-        let depth_store = Arc::new(DbDepthStore::new(db.clone(), noise(perf_params.header_data_cache_size)));
+        let daa_excluded_store = Arc::new(DbDaaStore::new(db.clone(), CachePolicy::Unit(noise(daa_excluded_cache_size))));
+        let headers_store = Arc::new(DbHeadersStore::new(
+            db.clone(),
+            CachePolicy::Tracked(noise(headers_cache_bytes)),
+            CachePolicy::Unit(noise((3600 * params.bps() as usize).max(perf_params.header_data_cache_size))),
+        ));
+        let depth_store = Arc::new(DbDepthStore::new(db.clone(), CachePolicy::Unit(noise(perf_params.header_data_cache_size))));
         let selected_chain_store =
-            Arc::new(RwLock::new(DbSelectedChainStore::new(db.clone(), noise(perf_params.header_data_cache_size))));
+            Arc::new(RwLock::new(DbSelectedChainStore::new(db.clone(), CachePolicy::Unit(noise(perf_params.header_data_cache_size)))));
 
         // Pruning
         let pruning_point_store = Arc::new(RwLock::new(DbPruningStore::new(db.clone())));
-        let past_pruning_points_store = Arc::new(DbPastPruningPointsStore::new(db.clone(), 4));
+        let past_pruning_points_store = Arc::new(DbPastPruningPointsStore::new(db.clone(), CachePolicy::Unit(1024)));
```

Comment:
Why did this change from 4 to 1024?

Reply:
They are simple hashes and might be accessed by `estimate_timestamp_from_daa_score` rpc calls. So I don't see a reason not to cover `~1000` points